src/core/modules/impl/ArweaveGQLTxsFetcher.ts

```diff
@@ -138,10 +137,26 @@ export class ArweaveGQLTxsFetcher {
     return response.edges[0].node;
   }
 
-  async transactions(variables: ArweaveTransactionQuery): Promise<GQLEdgeInterface[]> {
+  async transactions(
+    variables: ArweaveTransactionQuery,
+    pagesPerBatch: number,
+    signal?: AbortSignal
+  ): Promise<GQLEdgeInterface[]> {
     let pageResult = (await this.fetch<GQLResultInterface['data']>(TRANSACTIONS_QUERY, variables)).transactions;
     const edges: GQLEdgeInterface[] = [...pageResult.edges];
-    while (pageResult.pageInfo.hasNextPage) {
+    let pagesLoaded = 1;
+    if (pagesLoaded >= pagesPerBatch) {
+      return edges;
+    }
+    if (signal?.aborted) {
+      throw new Error(`Abort signal in ${ArweaveGQLTxsFetcher.name}`);
```

Comment:
Maybe this error should be of type `AbortError`?
similar to undici
https://github.com/nodejs/undici/blob/3fd7bf3a34f593c7d4fb03e48559755c659d1f1a/lib/fetch/index.js#L108C25-L108C25

Reply:
a dedicated AbortError has been added.